src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static newArr(animal) {
+    Animal.alive.push(animal);
+  }
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+  }
 }
-
+Animal.alive = [];
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden = false) {
+    super(name);
+
+    this.hidden = hidden;
+    Animal.newArr(this);
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    Animal.newArr(this);
+  }
+
+  bite(animal) {
+    if (animal.hidden === false) {
```

Comment:
Хижак може кусати тільки травоядних (а не інших хижаків). Потрібно додати перевірку ще й на це
```suggestion
    if (!animal.hidden) {
```